crates/typst/src/layout/length.rs

```diff
@@ -132,6 +132,32 @@ impl Length {
         self.ensure_that_em_is_zero(span, "inches")?;
         Ok(self.abs.to_inches())
     }
+
+    /// Resolve this length to an absolute length.
+    #[func]
+    pub fn resolve(
+        &self,
+        /// The styles with which to measure the length.
+        styles: Styles,
+    ) -> StrResult<Length> {
+        // Shortcircuit to avoid doing a stylechain lookup if we're already
+        // absolute.
+        if self.is_absolute() {
+            return Ok(*self);
+        }
+
+        let styles = StyleChain::new(&styles);
+        Ok(Length {
+            abs: self.abs + self.em.resolve(styles),
+            em: Em::zero(),
+        })
+    }
+
+    /// Returns whether this length is absolute (that is, has no `em` units).
+    #[func]
+    pub fn is_absolute(&self) -> bool {
+        self.em.is_zero()
+    }
```

Comment:
I don't think this method should be added. It's trivial to write in Typst, too, by accessing the `em` field.